### PR TITLE
Macro ignore extra semi colon and use anonymous exception passing

### PR DESCRIPTION
**Pull request by hjmjohnson**
Update exception macros to use anonymous exceptions for clang.

### CODE REVIEW

File: Modules/Core/Common/include/itkMacro.h [hunk 1]
```diff
@@ -120,9 +120,18 @@ namespace itk
 #  define ITK_GCC_PRAGMA_DIAG_PUSH() ITK_GCC_PRAGMA_DIAG(push)
 #  define ITK_GCC_PRAGMA_DIAG_POP() ITK_GCC_PRAGMA_DIAG(pop)
 #else
-#  define ITK_GCC_PRAGMA_DIAG(x)
-#  define ITK_GCC_PRAGMA_DIAG_PUSH()
-#  define ITK_GCC_PRAGMA_DIAG_POP()
+#  define ITK_GCC_PRAGMA_DIAG(x)                                                                                       \
```
Comment (blowekamp) [inline Modules/Core/Common/include/itkMacro.h]: It looks like "\" are missing here to me. If it is style compliant, putting it on one line may be clearer.

Comment (hjmjohnson) [reply Modules/Core/Common/include/itkMacro.h]: I don't understand about the "".  Do you mean the end of line continuations "\\"?  They are there, but off the standard github display.  You need to scrol to the right.

I did write these as one liners, but clang-format expanded it to 3 lines.  Perhaps this is a good use of a comment to ignore clang-format.

Comment (blowekamp) [reply Modules/Core/Common/include/itkMacro.h]: Sorry for the missing char! 😞 

I see the "\" now! That is very annoying that it is off the end of the GitHub display.

**Comment by dzenanz** [comment]
... and failed Windows build supports that :(

**Comment by hjmjohnson** [comment]
> I feel like there are more instances of `itkDebugMacro` and friends than the few which are part of this PR.

I'm not 100% sure what you are refering to, but I did make a mistake and added this paradigm in a pragma push/pop situation that it should not have been added to.  That is will be resolved in a force push in a few minutes.

**Pull request merged**